### PR TITLE
Show view statistics on resource page

### PR DESCRIPTION
**Pull request by rschenk**
Show a graph of views through time on resource pages. 

WorkVersions and Collections show the exact number of views for that particular resource.

Works show an aggregate of all _published_ versions.

File downloads are not shown at this time.

### CODE REVIEW

**Comment by awead** [comment]
@rschenk Looks good to me. Bypassing the ActiveRecord stuff really improves performance here, right? I'm assuming the main use case for this is powering the analytics graphs? I was wondering if there'd ever be the need to query a date range in the array of results and if another data structure might help with that. I can't think of one...

**Comment by rschenk** [comment]
@awead Yep that's exactly its intention is to power the graphs. Instantiating an ActiveRecord for each row would be expensive and pointless, since the graphs want to be fed a 2d-array _anyways,_ so why not cut out the middle man. 

This class can totally have date range filtering, and batched/page loading added to it later if need be

File: app/components/collection_metadata_component.rb [hunk 1]
```diff
@@ -16,7 +16,6 @@ class CollectionMetadataComponent < ActionView::Component::Base
     :title,
     :subtitle,
     :creator_aliases,
-    :description,
```
Comment (awead) [inline app/components/collection_metadata_component.rb:19]: @rschenk why did description get removed?

File: app/controllers/analytics_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    respond_to do |format|
+      format.html { head :unsupported_media_type }
+      format.json { render json: view_statistics }
+    end
+  end
+
+  private
+
+    def load_view_statistics(resource)
+      stats = if resource.is_a?(Work)
```
Comment (awead) [inline app/controllers/analytics_controller.rb:17]: @rschenk what if this was a method on the model? Work#stats and WorkVersion#stats

File: app/services/load_view_statistics.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+#   [Date, number of views that day, running total of all views through time]
+# @example
+#   > LoadViewStatistics::Aggregate.call(models: [model_1, model_2])
+#   => [
+#        [Date(2020-06-01), 1,     1],
+#        [Date(2020-06-02), 100, 101],
+#        [Date(2020-06-03), 3,   104]
+#      ]
+class LoadViewStatistics
+  class Aggregate
```
Comment (awead) [inline app/services/load_view_statistics.rb:63]: @rschenk This makes more sense to me as a separate file and named service, like `AggregateViewStatistics`. Alternatively, we can keep the class in the same file, but the Aggregate class ought to be inside the parent:

``` ruby
class LoadViewStatistics
  class Aggregate
  end
end
```

That way it looks more deliberate and not two different classes in the same file.

File: app/components/collection_metadata_component.rb (hunk 1 above)
Comment (rschenk) [reply app/components/collection_metadata_component.rb:19]: @awead Description is no longer shown in the metadata listing in the new UI design, it's shown in its own paragraph in the main content area of the view

File: app/controllers/analytics_controller.rb (hunk 2 above)
Comment (rschenk) [reply app/controllers/analytics_controller.rb:17]: I thought about that. Maybe I could put it inside the ViewStatistics concern module, then override it on Work

Comment (awead) [reply app/controllers/analytics_controller.rb:17]: meh. I'd say leave it for now. up to you...

**Pull request merged**